test/transport.spec.ts

```diff
@@ -284,7 +354,8 @@ describe("apiKMSKeyEnvVar", () => {
       },
     });
   });
-  it("does not set DD_KMS_API_KEY environment variable when flushMetricsToLogs is true", () => {
+
+  it("adds DD_KMS_API_KEY environment variable when flushMetricsToLogs is true", () => {
```

Comment:
Likewise here, I think we can just have one test case, the behavior is now the same when flushMetricsToLogs is true/false.